pdns/pdnsutil.cc

```diff
@@ -1599,6 +1599,8 @@ static int addOrReplaceRecord(bool addOrReplace, const vector<string>& cmds) {
   DNSName name;
   if (cmds.at(2) == "@")
     name=zone;
+  else if (boost::ends_with(cmds.at(2), "."))
```

Comment:
as @Habbie says, this can be `else if (isCanonical(cmds.at(2)))` ...

Reply:
Changed accordingly. Also added a commit to make use of `boost::ends_with()` in `isCanonical()`. Logically that is fine, I don't know if `isCanonical()` is performance-critical and how much worse `boost:ends_with()` is given that the suffix to test is a string and not a char.